knimin/handlers/barcode_util.py

```diff
@@ -258,34 +163,135 @@ def update_ag_barcode(self, barcode):
 """
-                body_message = body_message.format(name=login_user,
-                                                   barcode=barcode,
-                                                   sample_date=sample_date,
-                                                   sample_time=sample_time)
-            login_email = self.get_argument('login_email', None)
-            if login_email != "" or login_email is not None:
-                try:
-                    send_email(body_message, subject, login_email)
-                    sent_date = time.now()
-                    msg2 = ("Sent email successfully to kit owner %s"
-                            % login_email)
-                except:
-                    msg2 = ("Email sending to (%s) failed failed "
-                            "(barcode: %s)!!!<br/>" % (login_email, barcode))
-        sample_issue = self.get_argument('sample_issue', [])
-        moldy = overloaded = other = 'N'
-        if 'moldy' in sample_issue:
-            moldy = 'Y'
-        if 'overloaded' in sample_issue:
-            overloaded = 'Y'
-        if 'other' in sample_issue:
-            other = 'Y'
+            body_message = body_message.format(name=login_user,
+                                               barcode=barcode,
+                                               sample_date=sample_date,
+                                               sample_time=sample_time)
+
+        return subject, body_message
+
+
+class BarcodeUtilHandler(BaseHandler, BarcodeUtilHelper):
+    @authenticated
+    def get(self):
+        barcode = self.get_argument('barcode', None)
+        if barcode is None:
+            self.render("barcode_util.html", div_and_msg=None,
+                        barcode_projects=[], parent_project=None,
+                        project_names=[], barcode=None, email_type=None,
+                        barcode_info=None, proj_barcode_info=None, msgs=None,
+                        currentuser=self.current_user)
+            return
+        # gather info to display
+        barcode_details = db.get_barcode_details(barcode)
+        if len(barcode_details) == 0:
+            div_id = "invalid_barcode"
+            message = ("Barcode %s does not exist in the database" %
+                       barcode)
+            self.render("barcode_util.html",
+                        div_and_msg=(div_id, message, barcode),
+                        barcode_projects=[], parent_project=None,
+                        project_names=[],
+                        barcode=barcode, email_type=None,
+                        barcode_info=None, proj_barcode_info=None,
+                        msgs=None, currentuser=self.current_user)
+            return
+
+        barcode_projects, parent_project = db.getBarcodeProjType(
+            barcode)
+        project_names = db.getProjectNames()
+
+        # barcode exists get general info
+        if barcode_details['status'] is None:
+            barcode_details['status'] = 'Received'
```

Comment:
If it has no status, that means it has never been scanned before. Therefore, on first scan, it is now received.